workbench-core/environments/src/services/environmentTypeConfigService.test.ts

```diff
@@ -2,9 +2,33 @@
  *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  *  SPDX-License-Identifier: Apache-2.0
  */
-
-const envTypeConfigId = '40b01529-0c7f-4609-a1e2-715068da5f0e';
-jest.mock('uuid', () => ({ v4: () => envTypeConfigId }));
+const envTypeId = '1b0502f3-121f-4d63-b03a-44dc756e4c20';
+const envType = {
+  status: 'APPROVED',
+  createdAt: '2022-06-20T18:32:09.985Z',
+  updatedBy: 'owner-123',
+  createdBy: 'owner-123',
+  name: 'Jupyter Notebook',
+  resourceType: 'envType',
+  provisioningArtifactId: 'pa-dqwijdnwq12w2',
+  params: [],
+  updatedAt: '2022-06-20T18:36:14.358Z',
+  sk: `ET#${envTypeId}`,
+  owner: 'owner-123',
+  description: 'An Amazon SageMaker Jupyter Notebook',
+  id: envTypeId,
+  pk: `ET#${envTypeId}`,
+  productId: 'prod-dwqdqdqdwq2e3',
+  type: 'sagemaker'
+};
+const uuid = '40b01529-0c7f-4609-a1e2-715068da5f0e';
+const envTypeConfigId = `etc-${uuid}`;
+jest.mock('uuid', () => ({ v4: () => uuid }));
+jest.mock('./environmentTypeService', () => {
```

Comment:
I've seen Prototype mocking as well, which feels a little cleaner because everything doesn't need to be hoisted to the top of the file. Doing prototype mocks in the `beforeEach` for that `describe` also means that its scoped instead of global. Do you think it would make sense to replace the module mocking with prototype mocking here?

Reply:
added spy on method to receive same result